src/nrniv/nrnbbcore_write.cpp

```diff
@@ -2089,42 +271,75 @@ int nrncore_run(const char* arg) {
 
 /** Return neuron.coreneuron.enable */
 int nrncore_is_enabled() {
-  if (nrnpy_nrncore_enable_value_p_) {
-    int b = (*nrnpy_nrncore_enable_value_p_)();
-    return b;
-  }
-  return 0;
+    if (nrnpy_nrncore_enable_value_p_) {
+        int b = (*nrnpy_nrncore_enable_value_p_)();
+        return b;
+    }
+    return 0;
 }
 
 /** Run coreneuron with arg string from neuron.coreneuron.nrncore_arg(tstop)
  *  Return 0 on success
 */
 int nrncore_psolve(double tstop) {
-  if (nrnpy_nrncore_arg_p_) {
-    char* arg = (*nrnpy_nrncore_arg_p_)(tstop);
-    if (arg) {
-      nrncore_run(arg);
-      free(arg);
-      return 0;
+    if (nrnpy_nrncore_arg_p_) {
+        char* arg = (*nrnpy_nrncore_arg_p_)(tstop);
+        if (arg) {
+            nrncore_run(arg);
+            free(arg);
+            return 0;
+        }
     }
-  }
-  return -1;
+    return -1;
 }
 
 #else // !HAVE_DLFCN_H
 
-int nrncore_run(const char*) {
-  return -1;
+int nrncore_run(const char *) {
+    return -1;
 }
 
 int nrncore_is_enabled() {
-  return 0;
+    return 0;
 }
 
 int nrncore_psolve(double tstop) {
-  return 0;
+    return 0;
 }
 
 #endif //!HAVE_DLFCN_H
 
+// This function is related to stdindex2ptr in CoreNeuron to determine which values should
+// be transferred from CoreNeuron. Types correspond to the value to be transferred based on
+// mech_type enum or non-artificial cell mechanisms.
+// Limited to pointers to voltage, nt._nrn_fast_imem->_nrn_sav_rhs (fast_imem value) or
+// data of non-artificial cell mechanisms.
+// Requires cache_efficient mode.
+// Input double* and NrnThread. Output type and index.
+// type == 0 means could not determine index.
+int nrn_dblpntr2nrncore(double* pd, NrnThread& nt, int& type, int& index) {
```

Comment:
This function seems bit out of place in this file. can this be moved somewhere else?

Reply:
Will check, don't remember why I put it back

Reply:
Moved it.